bot/exts/evergreen/profile_pic_modification/pfp_modify.py

```diff
@@ -0,0 +1,241 @@
+import asyncio
+import json
+import logging
+import typing as t
+from concurrent.futures import ThreadPoolExecutor
+
+import aiofiles
+import discord
+from aiohttp import client_exceptions
+from discord.ext import commands
+
+from bot.constants import Colours
+from bot.exts.evergreen.profile_pic_modification._effects import PfpEffects
+from bot.utils.halloween import spookifications
+
+log = logging.getLogger(__name__)
+
+_EXECUTOR = ThreadPoolExecutor(10)
+
+
+async def in_executor(func: t.Callable, *args) -> t.Any:
+    """Allows non-async functions to work in async functions."""
+    log.trace(f"Running {func.__name__} in an executor.")
+    loop = asyncio.get_event_loop()
+    return await loop.run_in_executor(_EXECUTOR, func, *args)
+
+
+class PfpModify(commands.Cog):
+    """Various commands for users to change their own profile picture."""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        self.bot.loop.create_task(self.init_cog())
+
+    async def init_cog(self) -> None:
+        """Initial load from resources asynchronously."""
+        async with aiofiles.open('bot/resources/pride/gender_options.json') as f:
+            self.GENDER_OPTIONS = json.loads(await f.read())
+
+    @commands.group()
+    async def pfp_modify(self, ctx: commands.Context) -> None:
+        """Groups all of the pfp modifying commands to allow a single concurrency limit."""
+        if not ctx.invoked_subcommand:
+            await ctx.send_help(ctx.command)
+
+    @pfp_modify.command(name="8bitify", root_aliases=("8bitify",))
+    async def eightbit_command(self, ctx: commands.Context) -> None:
+        """Pixelates your avatar and changes the palette to an 8bit one."""
+        async with ctx.typing():
+            image_bytes = await ctx.author.avatar_url.read()
+            file = await in_executor(
+                PfpEffects.apply_effect,
+                image_bytes,
+                PfpEffects.eight_bitify_effect
+            )
+
+            embed = discord.Embed(
+                title="Your 8-bit avatar",
+                description="Here is your avatar. I think it looks all cool and 'retro'."
+            )
+
+            embed.set_image(url="attachment://modified_avatar.png")
```

Comment:
Can we give some more information here, like `eightbit_avatar_{user name}`. File name will be useful if the user wants to download the image for whatever reason.

Applicable for such cases.